engine/src/witness/sol.rs

```diff
@@ -175,6 +183,41 @@ impl VoterApi<SolanaLiveness> for SolanaLivenessVoter {
 	}
 }
 
+#[allow(dead_code)]
+#[derive(Clone)]
+struct SolanaVaultSwapsVoter {
+	client: SolRetryRpcClient,
+}
+
+#[async_trait::async_trait]
+impl VoterApi<SolanaVaultSwapTracking> for SolanaVaultSwapsVoter {
+	async fn vote(
+		&self,
+		settings: <SolanaVaultSwapTracking as ElectoralSystem>::ElectoralSettings,
+		properties: <SolanaVaultSwapTracking as ElectoralSystem>::ElectionProperties,
+	) -> Result<
+		<<SolanaVaultSwapTracking as ElectoralSystem>::Vote as VoteStorage>::Vote,
+		anyhow::Error,
+	> {
+		program_swaps_witnessing::get_program_swaps(
+			&self.client,
+			settings.swap_endpoint_data_account_address,
+			properties
+				.witnessed_open_accounts
+				.into_iter()
+				.map(|VaultSwapAccountAndSender { vault_swap_account, .. }| vault_swap_account)
+				.collect(),
```

Comment:
we can collect into a hashset here instead of a vec, and pass through a HashSet to these functions, we end up just doing `let sc_opened_accounts_hashset: HashSet<_> = sc_opened_accounts.iter().collect();` inside `get_changed_program_swap_accounts`